framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1280,6 +1280,8 @@ private void checkContractsAtMethodDeclaration(
           contract.viewpointAdaptDependentTypeAnnotation(
               atypeFactory, stringToJavaExpr, methodTree);
 
+      checkValidRefinementContract(annotation, contract.kind, methodTree);
```

Comment:
@mernst I understand that we initially discussed implementing this check in the implementation of `BaseTypeVisitor#visitAnnotation`, but I found that the contracts were already available in this method.

Let me know if this is an issue and I can go back to the original plan.

Reply:
As expected, I had to move the check into `BaseTypeVisitor#visitAnnotation`